File: backend/project-service/src/main/java/org/driventask/project/Payload/Request/ProjectRequest.java
package org.driventask.project.Payload.Request;

import java.time.LocalDateTime;
import java.util.Set;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record ProjectRequest(
    @NotNull
    @NotBlank
    String name,

    String description,
    
    LocalDateTime startDate,
    
    LocalDateTime endDate,
    
    Set<String> userId,
    
    Set<String> file_id
) {

}

File: frontend/src/app/models/project-request.ts
export interface ProjectRequest {
    id?: string;
    name: string;
    description?: string;
    startDate?: Date;
    endDate?: Date;
    status?: ProjectStatus;
    userId?: string[];
    file_id?: string[];
}

export enum ProjectStatus {
    Pending = 'Pending',
    InProgress = 'InProgress',
    Completed = 'Completed',
    Cancelled = 'Cancelled'
}